File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String[] log = logLine.replaceAll("[\\[:\\]]", "").trim().split(" ");
        String message = " ";
        for(int i = 1; i < log.length; i++){
            message += log[i] + " ";
        }
        return message.trim();
        //throw new UnsupportedOperationException("Please implement the (static) LogLevels.message() method");
    }

    public static String logLevel(String logLine) {
        String[] log = logLine.replaceAll("[\\[:\\]]", "").trim().split(" ");
        String level = log[0].toLowerCase();
        return level;
        //throw new UnsupportedOperationException("Please implement the (static) LogLevels.logLevel() method");
    }

    public static String reformat(String logLine) {
        String reformat = message(logLine) + " (" + logLevel(logLine) + ")";
        return reformat;
        //throw new UnsupportedOperationException("Please implement the (static) LogLevels.reformat() method");
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
using System.Text.RegularExpressions;

static class LogLine
{
    public static string Message(string logLine)
    {
        string message = logLine.Remove(0, logLine.IndexOf(":") + 1);
        return message.Trim();
        //throw new NotImplementedException("Please implement the (static) LogLine.Message() method");
    }

    public static string LogLevel(string logLine)
    {
        string level = logLine.Remove(logLine.IndexOf(":"));
        level = Regex.Replace(level, @"[\[\]]", "");
        return level.ToLower().Trim();
        //throw new NotImplementedException("Please implement the (static) LogLine.LogLevel() method");
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
        //throw new NotImplementedException("Please implement the (static) LogLine.Reformat() method");
    }
}
